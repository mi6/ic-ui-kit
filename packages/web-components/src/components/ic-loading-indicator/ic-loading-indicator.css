@import "../../global/normalize.css";

:host {
  /**
   * @prop --circular-diameter: Diameter of the circular-indicator
   */

  display: block;

  --linear-border-radius: 0.25rem;
  --inner-color: var(--ic-action-default);
  --outer-color: var(--ic-architectural-100);
  --label-color: var(--ic-color-primary-text);
  --margin: none;
  --linear-line-height: var(--ic-space-xs);
}

:host(.light) {
  --inner-color: var(--ic-architectural-white);
  --outer-color: var(--ic-architectural-800);
  --label-color: var(--ic-architectural-white);
}

.ic-loading-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

:host([size="small"]) {
  --circular-diameter: 2.5rem;
  --linear-line-height: var(--ic-space-xxs);
}

:host([size="default"]) {
  --circular-diameter: 5rem;
}

:host([size="large"]) {
  --circular-diameter: 7.5rem;
}

:host([size="icon"]) {
  display: inline-block;

  --margin: var(--ic-space-xxxs);
  --circular-diameter: 1.25rem;
}

:host(.label) {
  --margin: 0 0 var(--ic-space-xs) 0;
}

.ic-loading-label {
  white-space: nowrap;
  color: var(--label-color);
  margin-left: var(--label-margin-left);
}

.ic-loading-circular-outer {
  height: var(--circular-diameter);
  width: var(--circular-diameter);
}

.ic-loading-circular-outer.indeterminate {
  animation: circular-animation 1s linear;
  animation-iteration-count: infinite;
}

@keyframes circular-animation {
  100% {
    transform: rotate(360deg);
  }
}

.ic-loading-circular-outer,
.ic-loading-linear-outer {
  margin: var(--margin);
}

.ic-loading-linear-outer {
  background-color: var(--outer-color);
  height: var(--linear-line-height);
  width: 100%;
  border-radius: var(--linear-border-radius);
  overflow: hidden;
}

.ic-loading-linear-inner {
  position: relative;
  height: 100%;
  background-color: var(--inner-color);
  border-radius: var(--linear-border-radius);
}

.indeterminate > .ic-loading-linear-inner {
  animation: linear-animation 2s infinite;
}

.determinate > .ic-loading-linear-inner {
  transition: width 0.5s;
  width: var(--linear-width);
}

:host(.full-width) {
  --linear-border-radius: none;
}

@keyframes linear-animation {
  0% {
    width: 0%;
    left: -1%;
  }

  25% {
    width: 0%;
    left: -1%;
  }

  50% {
    width: 101%;
    left: -1%;
  }

  75% {
    width: 0%;
    left: 101%;
  }

  100% {
    width: 0%;
    left: 101%;
  }
}

.ic-loading-circular-svg {
  position: relative;
  width: 100%;
  height: 100%;
  transform: rotate(-90deg);
}

.ic-loading-circular-svg circle {
  width: 100%;
  height: 100%;
  fill: none;
  stroke: var(--outer-color);
  stroke-width: var(--circular-line-width);
  stroke-linecap: round;
}

/* --stroke-dash-array calculated via dashArray method */
.ic-loading-circular-svg circle:nth-child(2) {
  --circular-indeterminate: calc(
    (0.25 * var(--stroke-dasharray)) - var(--stroke-dasharray)
  );

  stroke-dasharray: var(--stroke-dasharray), var(--stroke-dasharray);
  stroke-dashoffset: var(--stroke-dashoffset, var(--circular-indeterminate));
  stroke: var(--compact-step-inner-color, var(--inner-color));
}

:host(.inner-label) .ic-loading-circular-inner {
  display: grid;
}

:host(.inner-label) .ic-loading-circular-svg,
.inner-text {
  grid-column: 1;
  grid-row: 1;
}

:host(.not-required.compact-step-progress-indicator)
  .ic-loading-circular-svg
  circle:nth-child(2) {
  stroke: var(--ic-architectural-300);
}

.inner-text {
  display: flex;
  align-items: center;
  justify-content: center;
  width: var(--ic-space-lg);
  height: inherit;
  overflow: hidden;
  margin: auto;
  color: var(--ic-status-info);
}

:host(.not-required.compact-step-progress-indicator) .inner-text {
  color: var(--ic-color-tertiary-text);
}

/** High Contrast **/
@media (forced-colors: active) {
  .indeterminate > .ic-loading-circular-inner {
    forced-color-adjust: none;
  }

  .ic-loading-linear-outer {
    border: var(--ic-hc-border);
  }

  .ic-loading-linear-inner {
    background-color: canvastext;
  }

  .ic-loading-circular-svg circle {
    stroke: Background;
  }

  .ic-loading-circular-svg circle:nth-child(2) {
    stroke: canvastext;
  }

  :host(.not-required.compact-step-progress-indicator)
    .ic-loading-circular-svg
    circle:nth-child(2) {
    stroke: GrayText;
  }
}
