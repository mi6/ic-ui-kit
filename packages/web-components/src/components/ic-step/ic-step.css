/* SHARED STYLING */
:host {
  display: flex;
  flex: auto;
}

.step {
  display: flex;
  flex: 1 1 0;
}

.step-icon {
  display: flex;
  justify-content: center;
}

.step-title,
.step-subtitle,
.step-status,
.next-step {
  white-space: pre-line;
}

.visually-hidden {
  position: absolute;
  left: -625rem;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

/* COMPACT STEP STYLING */
:host(.compact) {
  column-gap: var(--ic-space-sm);

  --compact-step-inner-color: var(--ic-status-info);
  --compact-step-circular-line-width: var(--ic-space-xxs);
}

:host(.compact) .step {
  column-gap: var(--ic-space-sm);
}

:host(.compact) .step:not(.current) {
  display: none;
  opacity: 0;
  visibility: hidden;
}

.compact-step-progress-indicator {
  margin: var(--ic-space-xs) 0 0;
}

:host(.compact) .step-title-area {
  display: flex;
  flex-direction: column;
  width: 14.25rem;
}

.info-line {
  display: flex;
  column-gap: var(--ic-space-xs);
}

.step-status {
  display: flex;
  color: var(--ic-color-secondary-text);
  column-gap: var(--ic-space-xxxs);
}

.compact-step-completed .step-status {
  color: var(--ic-status-success);
}

.compact-step-disabled :is(.step-title, .step-status) {
  color: var(--ic-color-tertiary-text);
}

.step-num {
  color: var(--ic-color-secondary-text);
  white-space: nowrap;
}

:host(.compact) .step-icon {
  margin: var(--ic-space-xxxs);
}

:host(.compact) .step-icon svg {
  width: var(--ic-space-md);
  height: var(--ic-space-md);
}

/* DEFAULT STEP STYLING */
:host(.default) .step {
  flex-direction: column;
}

:host(.default.last-step) {
  flex-grow: initial;
}

.step-top {
  display: flex;
  width: 100%;
  align-items: center;
  align-self: flex-start;
  height: 2.5rem;
}

:host(.default) .step-icon {
  border-radius: 50%;
}

.step-icon-inner {
  width: var(--ic-space-xl);
  height: var(--ic-space-xl);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
}

:host(.default) .current {
  color: var(--ic-status-info);
}

:host(.default) .step-title-area {
  margin: var(--ic-space-xs) 0;
  padding-right: var(--ic-space-xs);
  width: 100%;
}

:host(.default) .step-title,
.step-subtitle {
  width: fit-content;
}

.step-subtitle {
  color: var(--ic-color-tertiary-text);
}

.current .step-subtitle {
  color: var(--ic-color-primary-text);
}

:host(.default) .completed {
  color: var(--ic-status-success);
}

.active .step-icon-inner {
  box-shadow: inset var(--ic-architectural-200) 0 0 0 0.125rem;
}

.current .step-icon-inner {
  background-color: var(--ic-status-info);
  color: white;
}

.disabled {
  color: var(--ic-architectural-200);
}

.disabled .step-icon-inner {
  border: var(--ic-space-1px) dashed var(--ic-architectural-200);
  width: calc(var(--ic-space-xl) - var(--ic-space-xxxs));
  height: calc(var(--ic-space-xl) - var(--ic-space-xxxs));
}

.disabled .step-title-area {
  color: var(--ic-color-tertiary-text);
}

.completed .step-icon-inner {
  background: var(--ic-status-success);
  box-shadow: inset var(--ic-status-success) 0 0 0 var(--ic-space-xxxs);
  border-radius: 100%;
}

:host(.default) .current .step-icon {
  border: var(--ic-space-xxxs) solid var(--ic-status-info);
  padding: var(--ic-space-xxxs);
  margin: 0 calc(-1 * var(--ic-space-xxxs));

  /* compensating for the circle being bigger than other steps */
}

.step-connect {
  height: var(--ic-space-xxxs);
  background-color: var(--ic-architectural-200);
  margin: 0 var(--ic-space-xs);
  border-radius: var(--ic-space-xxs);
  width: 100%;
}

.aligned-full-width.step-connect {
  min-width: 6.25rem;
  width: 100%;
}

.disabled .step-connect {
  height: 0;
  background-color: rgb(0 0 0 / 0%);
  border-top: 0.125rem dashed var(--ic-architectural-200);
  border-radius: 0;
}

.completed .step-connect {
  background-color: var(--ic-status-success);
}

.step-connect-inner {
  width: 70%;
  display: flex;
  flex: auto;
  height: var(--ic-space-xxxs);
  border-radius: var(--ic-space-xxs);
  background-color: var(--ic-status-info);
}

.step-icon-inner .check-icon {
  padding-top: var(--ic-space-xxs);
}

.step-icon-inner .check-icon svg {
  width: var(--ic-space-md);
  height: auto;
}

.step-icon-inner .check-icon > svg > path {
  fill: var(--ic-color-white-text);
}

/** High Contrast **/
@media (forced-colors: active) {
  /* COMPACT STEP */
  .compact-step-disabled :is(.step-title, .step-status) {
    color: GrayText;
  }

  /* DEFAULT STEP */
  .step-connect:not(.disabled .step-connect) {
    border: var(--ic-hc-border);
  }

  .active .step-icon-inner,
  .completed .step-icon-inner,
  .current .step-icon-inner {
    forced-color-adjust: none;
    box-shadow: inset canvastext 0 0 0 0.125rem;
    background-color: transparent;
    color: canvastext;
  }

  :host(.default) .current .step-icon {
    padding: 0;
    border: none;
  }

  .disabled,
  .disabled .step-title-area {
    color: GrayText;
  }

  .step-connect-inner,
  .completed .step-connect {
    background-color: canvastext;
  }

  .step-icon-inner .check-icon > svg > path {
    fill: canvastext;
  }
}
