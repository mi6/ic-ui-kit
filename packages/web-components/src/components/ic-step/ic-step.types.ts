export type IcStepVariants = "default" | "compact";

export type IcStepStatuses = "optional" | "required";

export type IcStepTypes = "active" | "current" | "completed" | "disabled";
