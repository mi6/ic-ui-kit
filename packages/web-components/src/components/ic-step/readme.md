# ic-step



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description                                                                                                                                             | Type                                                 | Default     |
| -------------- | --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- | ----------- |
| `stepStatus`   | `step-status`   | The status of the step. Use this prop to display a status message on the step if it is required or optional.                                            | `"optional" \| "required"`                           | `undefined` |
| `stepSubtitle` | `step-subtitle` | Additional information about the step. Use this prop to override the default step status messaging displayed when selecting a step type or step status. | `string`                                             | `undefined` |
| `stepTitle`    | `step-title`    | The title of the step within the stepper.                                                                                                               | `string`                                             | `undefined` |
| `stepType`     | `step-type`     | The state of the step within the stepper.                                                                                                               | `"active" \| "completed" \| "current" \| "disabled"` | `"active"`  |


## Dependencies

### Depends on

- [ic-loading-indicator](../ic-loading-indicator)
- [ic-typography](../ic-typography)

### Graph
```mermaid
graph TD;
  ic-step --> ic-loading-indicator
  ic-step --> ic-typography
  ic-loading-indicator --> ic-typography
  style ic-step fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


