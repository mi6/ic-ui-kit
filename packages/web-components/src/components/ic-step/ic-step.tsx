import { Component, Host, h, Prop, Element, Watch } from "@stencil/core";
import checkIcon from "../../assets/check-icon.svg";
import warningIcon from "../../assets/warning-icon-outline.svg";
import { IcStepVariants, IcStepStatuses, IcStepTypes } from "./ic-step.types";

@Component({
  tag: "ic-step",
  styleUrl: "ic-step.css",
  shadow: true,
})
export class Step {
  @Element() el: HTMLIcStepElement;

  /**
   * @internal If a compact stepper is being used, this sets the styling of the step.
   */
  @Prop() compactStepStyling?: IcStepTypes;

  /**
   * @internal If `true`, and a compact stepper  is being used, the current step will be the only step in view.
   */
  @Prop({ mutable: true }) current?: boolean = false;

  /**
   * @internal If `true`, the step will be marked as being the last one in the series. This is managed by ic-stepper.
   */
  @Prop() lastStep!: boolean;

  /**
   * @internal The step number of the final step. This is managed by ic-stepper.
   */
  @Prop() lastStepNum?: number;

  /**
   * @internal The name of the next step. This is managed by ic-stepper.
   */
  @Prop() nextStepTitle?: string;

  /**
   * @internal The progress of the next step, calculated by dividing the current step number by the total number of steps. This is managed by ic-stepper.
   */
  @Prop() progress?: number;

  /**
   * @internal The step number, managed by ic-stepper.
   */
  @Prop() stepNum?: number;

  /**
   * The status of the step. Use this prop to display a status message on the step if it is required or optional.
   */
  @Prop() stepStatus?: IcStepStatuses;

  /**
   * Additional information about the step. Use this prop to override the default step status messaging displayed when selecting a step type or step status.
   */
  @Prop() stepSubtitle?: string;

  /**
   * The title of the step within the stepper.
   */
  @Prop() stepTitle?: string;

  /**
   * @internal The variant of the step. This is managed by ic-stepper.
   */
  @Prop() variant!: IcStepVariants;

  /**
   * The state of the step within the stepper.
   */
  @Prop() stepType?: IcStepTypes = "active";

  @Watch("stepType")
  stepTypeChangeHandler(): void {
    if (this.variant === "compact" && this.stepType === "current") {
      this.current = true;
    } else {
      this.current = false;
    }
  }

  render() {
    // ARIA LABEL
    let ariaLabel = "";

    if (this.stepType === "completed") {
      ariaLabel = ". Completed step";
    } else if (this.stepType === "disabled") {
      ariaLabel = ". Non-required step";
    } else if (this.stepStatus === "required") {
      ariaLabel = ". Required step";
    } else if (this.stepStatus === "optional") {
      ariaLabel = ". Optional step";
    }

    // STEP STATUS
    let stepStatus;
    if (this.stepStatus) {
      stepStatus = this.stepStatus[0].toUpperCase() + this.stepStatus.slice(1);
    }

    // STEP TYPE
    let stepType;
    if (
      this.stepType === "disabled" ||
      this.compactStepStyling === "disabled"
    ) {
      stepType = "Not required";
    } else if (this.compactStepStyling === "completed") {
      stepType = "Completed";
    }

    // STATUS ICON FOR COMPACT STEP
    let statusIcon;
    if (
      this.stepType === "completed" ||
      this.compactStepStyling === "completed"
    ) {
      statusIcon = (
        <span
          class="check-icon step-icon"
          aria-hidden="true"
          innerHTML={checkIcon}
        ></span>
      );
    } else if (
      this.stepType === "disabled" ||
      this.compactStepStyling === "disabled"
    ) {
      statusIcon = (
        <span
          class="warning-icon step-icon"
          aria-hidden="true"
          innerHTML={warningIcon}
        ></span>
      );
    }

    // COMPACT STEP COMPONENT
    const compactStep = (
      <div
        class={{
          ["step"]: true,
          ["current"]: this.current,
          [`compact-step-${this.compactStepStyling}`]:
            !!this.compactStepStyling,
        }}
      >
        <ic-loading-indicator
          class={{
            "compact-step-progress-indicator": true,
            "not-required":
              this.stepType === "disabled" ||
              this.compactStepStyling === "disabled",
          }}
          aria-hidden="true"
          size="small"
          inner-label={this.stepNum}
          progress={this.progress}
        ></ic-loading-indicator>
        <div class="step-title-area">
          <ic-typography variant="h4" class="step-title">
            {this.stepTitle}
          </ic-typography>
          <div class="info-line">
            <ic-typography variant="caption" class="step-num">
              {`${this.stepNum} of ${this.lastStepNum}`}
              <span class="visually-hidden"> steps</span>
            </ic-typography>
            {(this.stepSubtitle ||
              this.stepType === "completed" ||
              this.stepType === "disabled" ||
              (this.variant === "compact" &&
                !!this.compactStepStyling &&
                this.compactStepStyling !== "active") ||
              !!this.stepStatus) && (
              <div class="step-status">
                {statusIcon !== undefined && statusIcon}
                {(this.stepSubtitle || stepType) && (
                  <ic-typography variant="caption">
                    {this.stepSubtitle !== null &&
                    this.stepSubtitle !== undefined
                      ? this.stepSubtitle
                      : this.stepType === "disabled" ||
                        (this.variant === "compact" &&
                          this.compactStepStyling === "disabled") ||
                        this.stepType === "completed" ||
                        (this.variant === "compact" &&
                          this.compactStepStyling === "completed")
                      ? stepType
                      : this.stepStatus && stepStatus}
                  </ic-typography>
                )}
              </div>
            )}
          </div>
          {this.lastStep ? (
            <ic-typography variant="subtitle-small" class="next-step">
              Last step
            </ic-typography>
          ) : (
            this.nextStepTitle !== undefined && (
              <ic-typography variant="subtitle-small" class="next-step">
                Next<span class="visually-hidden"> step is</span>:{" "}
                {this.nextStepTitle}
              </ic-typography>
            )
          )}
        </div>
      </div>
    );

    // ICON FOR DEFAULT STEP
    let icon;
    if (this.stepType !== "completed") {
      icon = (
        <ic-typography variant="subtitle-small">
          <span class="step-icon-inner" aria-hidden="true">
            {this.stepNum}
          </span>
        </ic-typography>
      );
    } else {
      icon = (
        <div class="step-icon-inner" aria-hidden="true">
          <span class="check-icon" innerHTML={checkIcon}></span>
        </div>
      );
    }

    // STEP CONNECT FOR DEFAULT STEP
    const partialBar = this.stepType === "current" && (
      <div class="step-connect-inner"></div>
    );

    const finalStep = !this.lastStep && (
      <div
        class={{
          ["step-connect"]: true,
          ["aligned-full-width"]:
            this.el.parentElement.classList.contains("default") &&
            !this.el.parentElement.classList.contains("aligned-left"),
        }}
      >
        {partialBar}
      </div>
    );

    // DEFAULT STEP COMPONENT
    const defaultStep = (
      <div
        class={{
          ["step"]: true,
          [`${this.stepType}`]: true,
        }}
      >
        <div class="step-top">
          <div class="step-icon">{icon}</div>
          {finalStep}
        </div>
        {(this.stepTitle || this.stepSubtitle || this.stepStatus) && (
          <div class="step-title-area">
            {this.stepTitle && (
              <ic-typography variant="subtitle-large" class="step-title">
                {this.stepTitle}
              </ic-typography>
            )}
            {this.stepTitle && (this.stepSubtitle || this.stepStatus) && (
              <ic-typography variant="caption" class="step-subtitle">
                {this.stepSubtitle !== null && this.stepSubtitle !== undefined
                  ? this.stepSubtitle
                  : stepStatus}
              </ic-typography>
            )}
          </div>
        )}
      </div>
    );

    return (
      <Host
        role="listitem"
        aria-label={`Step ${this.stepNum}${ariaLabel}`}
        aria-current={(this.current || this.stepType === "current") && "step"}
        class={{
          ["aligned-full-width"]:
            this.el.parentElement.classList.contains("default") &&
            !this.el.parentElement.classList.contains("aligned-left"),
          [`${this.variant}`]: true,
        }}
      >
        {this.variant === "compact" ? compactStep : defaultStep}
      </Host>
    );
  }
}
