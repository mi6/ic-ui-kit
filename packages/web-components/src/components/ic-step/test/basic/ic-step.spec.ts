import { newSpecPage } from "@stencil/core/testing";
import { Step } from "../../ic-step";
import { Typography } from "../../../ic-typography/ic-typography";
import { LoadingIndicator } from "../../../ic-loading-indicator/ic-loading-indicator";

describe("default variant of ic-step component", () => {
  it("should render default step with no title", async () => {
    const page = await newSpecPage({
      components: [Step, Typography],
      html: `<ic-step variant="default"></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it("should render default step with correct title", async () => {
    const page = await newSpecPage({
      components: [Step, Typography],
      html: `<ic-step variant="default" step-title="First"></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render default step with correct title, subtitle and current state", async () => {
    const page = await newSpecPage({
      components: [Step, Typography],
      html: `<ic-step variant="default" 
      step-title="Second With a Very Long Title"
      step-subtitle="Optional Subtitle"
      step-type="current"
    ></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render disabled stepType", async () => {
    const page = await newSpecPage({
      components: [Step, Typography],
      html: `<ic-step variant="default" 
      step-type="disabled"
    ></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render completed stepType", async () => {
    const page = await newSpecPage({
      components: [Step, Typography],
      html: `<ic-step variant="default" 
      step-type="completed"
    ></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});

describe("compact variant of ic-step component", () => {
  it("should render a compact step with no step status", async () => {
    const page = await newSpecPage({
      components: [Step],
      html: `<ic-step
      variant="compact" 
      step-type="current"
      last-step="false"
      step-num="3"
      last-step-num="4"
      progress="75"
      step-title="Third step"
      next-step-title="Fourth step"></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render an step status icon for a step with a disabled stepType", async () => {
    const page = await newSpecPage({
      components: [Step],
      html: `<ic-step
      variant="compact"
      step-type="disabled"
      compact-step-styling = "disabled"
      last-step="false"
      step-num="3"
      last-step-num="4"
      progress="75"
      step-title="Third step"
      next-step-title="next step"
    ></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();

    const stepStatus = page.root.shadowRoot.querySelector(
      ".step-status > ic-typography"
    ).textContent;

    expect(stepStatus).toBe("Not required");

    const statusIcon = page.root.shadowRoot.querySelector("span.step-icon");

    expect(statusIcon).toEqualHtml(
      `<span class="warning-icon step-icon" aria-hidden="true">svg</span>`
    );
  });

  it("should render a step status icon for a step with a completed stepType", async () => {
    const page = await newSpecPage({
      components: [Step],
      html: `<ic-step
      variant="compact"
      last-step="false"
      step-num="3"
      last-step-num="4"
      progress="75"
      step-title="Third step"
      next-step-title="next step"
      step-type="completed"
      compact-step-styling = "completed"
    ></ic-step>`,
    });

    expect(page.root).toMatchSnapshot();

    const stepStatus = page.root.shadowRoot.querySelector(
      ".step-status > ic-typography"
    ).textContent;

    expect(stepStatus).toBe("Completed");

    const statusIcon = page.root.shadowRoot.querySelector("span.step-icon");

    expect(statusIcon).toEqualHtml(
      `<span class="check-icon step-icon" aria-hidden="true">svg</span>`
    );
  });

  it("should set next step text as last step if last step is true", async () => {
    const page = await newSpecPage({
      components: [Step],
      html: `<ic-step
      variant="compact" 
      step-type="current"
      last-step="true"
      step-num=4"
      last-step-num="4"
      progress="100"
      step-title="Final step"
    ></ic-step>`,
    });

    const lastStep =
      page.root.shadowRoot.querySelector(".next-step").textContent;

    expect(lastStep).toBe("Last step");
  });

  it("should set the circular line width of the loading indicator as 40 for all steps", async () => {
    jest.spyOn(console, "warn").mockImplementation(jest.fn());
    const page = await newSpecPage({
      components: [Step, LoadingIndicator],
      html: `<ic-step
      variant="compact" 
      last-step="true"
      step-num=4"
      last-step-num="4"
      progress="100"
      step-title="Final step"
    ></ic-step>`,
    });

    const loadingIndicator = page.root.shadowRoot.querySelector(
      ".step > ic-loading-indicator"
    );

    expect(
      loadingIndicator.classList.contains("compact-step-progress-indicator")
    ).toBeTruthy();
  });

  it("should watch the stepType & change the value of the current prop if stepType = 'current'", async () => {
    const page = await newSpecPage({
      components: [Step],
      html: `<ic-step
      variant="compact" 
      last-step="true"
      step-num=4"
      last-step-num="4"
      progress="100"
      step-title="Final step"
      step-type = "current"
    ></ic-step>`,
    });

    await page.rootInstance.stepTypeChangeHandler();
    await page.waitForChanges();
    expect(page.rootInstance.current).toBeTruthy();
  });

  it("should watch the stepType & shouldn't change the value of the current prop if stepType is not 'current'", async () => {
    const page = await newSpecPage({
      components: [Step],
      html: `<ic-step
      variant="compact" 
      last-step="true"
      step-num=4"
      last-step-num="4"
      progress="100"
      step-title="Final step"
      step-type = "completed"
      compact-step-styling = "completed"
    ></ic-step>`,
    });

    await page.rootInstance.stepTypeChangeHandler();
    await page.waitForChanges();
    expect(page.rootInstance.current).toBeFalsy();
  });
});
