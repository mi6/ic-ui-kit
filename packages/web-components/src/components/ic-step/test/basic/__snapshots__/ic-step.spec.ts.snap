// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compact variant of ic-step component should render a compact step with no step status 1`] = `
<ic-step aria-current="step" aria-label="Step 3" class="compact" last-step="false" last-step-num="4" next-step-title="Fourth step" progress="75" role="listitem" step-num="3" step-title="Third step" step-type="current" variant="compact">
  <mock:shadow-root>
    <div class="step">
      <ic-loading-indicator aria-hidden="true" class="compact-step-progress-indicator" inner-label="3" progress="75" size="small"></ic-loading-indicator>
      <div class="step-title-area">
        <ic-typography class="step-title" variant="h4">
          Third step
        </ic-typography>
        <div class="info-line">
          <ic-typography class="step-num" variant="caption">
            3 of 4
            <span class="visually-hidden">
              steps
            </span>
          </ic-typography>
        </div>
        <ic-typography class="next-step" variant="subtitle-small">
          Next
          <span class="visually-hidden">
            step is
          </span>
          : Fourth step
        </ic-typography>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`compact variant of ic-step component should render a step status icon for a step with a completed stepType 1`] = `
<ic-step aria-label="Step 3. Completed step" class="compact" compact-step-styling="completed" last-step="false" last-step-num="4" next-step-title="next step" progress="75" role="listitem" step-num="3" step-title="Third step" step-type="completed" variant="compact">
  <mock:shadow-root>
    <div class="compact-step-completed step">
      <ic-loading-indicator aria-hidden="true" class="compact-step-progress-indicator" inner-label="3" progress="75" size="small"></ic-loading-indicator>
      <div class="step-title-area">
        <ic-typography class="step-title" variant="h4">
          Third step
        </ic-typography>
        <div class="info-line">
          <ic-typography class="step-num" variant="caption">
            3 of 4
            <span class="visually-hidden">
              steps
            </span>
          </ic-typography>
          <div class="step-status">
            <span aria-hidden="true" class="check-icon step-icon">
              svg
            </span>
            <ic-typography variant="caption">
              Completed
            </ic-typography>
          </div>
        </div>
        <ic-typography class="next-step" variant="subtitle-small">
          Next
          <span class="visually-hidden">
            step is
          </span>
          : next step
        </ic-typography>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`compact variant of ic-step component should render an step status icon for a step with a disabled stepType 1`] = `
<ic-step aria-label="Step 3. Non-required step" class="compact" compact-step-styling="disabled" last-step="false" last-step-num="4" next-step-title="next step" progress="75" role="listitem" step-num="3" step-title="Third step" step-type="disabled" variant="compact">
  <mock:shadow-root>
    <div class="compact-step-disabled step">
      <ic-loading-indicator aria-hidden="true" class="compact-step-progress-indicator not-required" inner-label="3" progress="75" size="small"></ic-loading-indicator>
      <div class="step-title-area">
        <ic-typography class="step-title" variant="h4">
          Third step
        </ic-typography>
        <div class="info-line">
          <ic-typography class="step-num" variant="caption">
            3 of 4
            <span class="visually-hidden">
              steps
            </span>
          </ic-typography>
          <div class="step-status">
            <span aria-hidden="true" class="step-icon warning-icon">
              svg
            </span>
            <ic-typography variant="caption">
              Not required
            </ic-typography>
          </div>
        </div>
        <ic-typography class="next-step" variant="subtitle-small">
          Next
          <span class="visually-hidden">
            step is
          </span>
          : next step
        </ic-typography>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`default variant of ic-step component should render completed stepType 1`] = `
<ic-step aria-label="Step undefined. Completed step" class="default" role="listitem" step-type="completed" variant="default">
  <mock:shadow-root>
    <div class="completed step">
      <div class="step-top">
        <div class="step-icon">
          <div aria-hidden="true" class="step-icon-inner">
            <span class="check-icon">
              svg
            </span>
          </div>
        </div>
        <div class="step-connect"></div>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`default variant of ic-step component should render default step with correct title 1`] = `
<ic-step aria-label="Step undefined" class="default" role="listitem" step-title="First" variant="default">
  <mock:shadow-root>
    <div class="active step">
      <div class="step-top">
        <div class="step-icon">
          <ic-typography class="ic-typography-subtitle-small">
            <mock:shadow-root>
              <slot></slot>
            </mock:shadow-root>
            <span aria-hidden="true" class="step-icon-inner"></span>
          </ic-typography>
        </div>
        <div class="step-connect"></div>
      </div>
      <div class="step-title-area">
        <ic-typography class="ic-typography-subtitle-large step-title">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          First
        </ic-typography>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`default variant of ic-step component should render default step with correct title, subtitle and current state 1`] = `
<ic-step aria-current="step" aria-label="Step undefined" class="default" role="listitem" step-subtitle="Optional Subtitle" step-title="Second With a Very Long Title" step-type="current" variant="default">
  <mock:shadow-root>
    <div class="current step">
      <div class="step-top">
        <div class="step-icon">
          <ic-typography class="ic-typography-subtitle-small">
            <mock:shadow-root>
              <slot></slot>
            </mock:shadow-root>
            <span aria-hidden="true" class="step-icon-inner"></span>
          </ic-typography>
        </div>
        <div class="step-connect">
          <div class="step-connect-inner"></div>
        </div>
      </div>
      <div class="step-title-area">
        <ic-typography class="ic-typography-subtitle-large step-title">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          Second With a Very Long Title
        </ic-typography>
        <ic-typography class="ic-typography-caption step-subtitle">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          Optional Subtitle
        </ic-typography>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`default variant of ic-step component should render default step with no title 1`] = `
<ic-step aria-label="Step undefined" class="default" role="listitem" variant="default">
  <mock:shadow-root>
    <div class="active step">
      <div class="step-top">
        <div class="step-icon">
          <ic-typography class="ic-typography-subtitle-small">
            <mock:shadow-root>
              <slot></slot>
            </mock:shadow-root>
            <span aria-hidden="true" class="step-icon-inner"></span>
          </ic-typography>
        </div>
        <div class="step-connect"></div>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;

exports[`default variant of ic-step component should render disabled stepType 1`] = `
<ic-step aria-label="Step undefined. Non-required step" class="default" role="listitem" step-type="disabled" variant="default">
  <mock:shadow-root>
    <div class="disabled step">
      <div class="step-top">
        <div class="step-icon">
          <ic-typography class="ic-typography-subtitle-small">
            <mock:shadow-root>
              <slot></slot>
            </mock:shadow-root>
            <span aria-hidden="true" class="step-icon-inner"></span>
          </ic-typography>
        </div>
        <div class="step-connect"></div>
      </div>
    </div>
  </mock:shadow-root>
</ic-step>
`;
